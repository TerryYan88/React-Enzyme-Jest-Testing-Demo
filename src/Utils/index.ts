/* eslint-disable react/forbid-foreign-prop-types */
import {ShallowWrapper} from "enzyme";
import checkPropTypes from "check-prop-types";
import { applyMiddleware,createStore } from "redux";
import { rootReducer } from "../redux/reducers/rootReducer";


export const testStore = (initialState={}) =>{
    const createStoreWithMiddleware = applyMiddleware()(createStore)
    return createStoreWithMiddleware(rootReducer,initialState);
}


export const findByTestAttr = (component: ShallowWrapper,attr:string)=>{
    const wrapper = component.find(`[data-test="${attr}"]`);
    return wrapper;
}

export const checkProps = (component:any,expectedProps:{})=>{
    const propsErr = checkPropTypes(component.propTypes,expectedProps,'props',component.name);
    return propsErr;
}


