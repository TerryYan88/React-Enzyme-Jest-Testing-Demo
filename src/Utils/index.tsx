
import {ShallowWrapper} from "enzyme";


export const findByTestAttr = (component: ShallowWrapper,attr:string)=>{
    const wrapper = component.find(`[data-test="${attr}"]`);
    return wrapper;
}