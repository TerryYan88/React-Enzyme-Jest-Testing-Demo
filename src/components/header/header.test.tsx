import React from 'react';
import {shallow,ShallowWrapper} from "enzyme";
import Header from "./header";
import {HeaderContainer} from "./headerStyle";
import {findByTestAttr} from "../../Utils/index"

const setUp = (props={})=>{
    const component = shallow(<Header {...props}/>) // can test props 
    return component;
}

describe('Header Component', () => {
    let component:ShallowWrapper;
    //beforeEach(): before every single test 
    //all declarations should pass function  const component = shallow(<Header {...props}/>) first
    beforeEach(() => { 
        component = setUp();
        
    })

    it("should render without errors",()=>{  
        const container = component.find(HeaderContainer);
        expect(container.length).toBe(1);
        const wrapper = findByTestAttr(component,"headerComponent");
        expect(wrapper.length).toBe(1);
    })
    it("should render a logo",()=>{
        //data-test 
        const logo = findByTestAttr(component,"logoImg");
        expect(logo.length).toBe(1);
    })

})