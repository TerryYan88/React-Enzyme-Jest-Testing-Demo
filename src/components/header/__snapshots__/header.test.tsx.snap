// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component snapshot 1`] = `
.c0 {
  display: block;
  width: 100%;
  height: 50px;
  margin: 0 auto;
  padding: 0 10px;
  background-color: black;
}

.c0 .wrap {
  position: relative;
  display: block;
  width: 100%;
  height: 100%;
  margin: 0 auto;
  max-width: 980px;
}

.c0 .wrap .logo {
  display: block;
  width: 250px;
  position: absolute;
  left: 10px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c0 .wrap .logo img {
  width: 20%;
  margin: 0;
}

<div
  className="c0"
  data-test="headerComponent"
>
  <div
    className="wrap"
  >
    <div
      className="logo"
    >
      <img
        alt="Logo"
        data-test="logoImg"
        src="http://localhost:3000/assets/2.png"
      />
    </div>
  </div>
</div>
`;
